server/src/main/java/org/apache/iotdb/db/tools/DiskSpiller.java

```diff
@@ -0,0 +1,191 @@
+    index++;
+
+    ListenableFuture<Boolean> future =
+        Futures.submit(() -> writeData(tsBlocks, fileName), directExecutor());
+    processingTask.add(future);
+  }
+
+  public void spillSortedData(List<MergeSortKey> sortedData) throws IOException {
+    List<TsBlock> tsBlocks = new ArrayList<>();
+    TsBlockBuilder tsBlockBuilder = new TsBlockBuilder(dataTypeList);
+    ColumnBuilder[] columnBuilders = tsBlockBuilder.getValueColumnBuilders();
+    ColumnBuilder timeColumnBuilder = tsBlockBuilder.getTimeColumnBuilder();
+
+    for (MergeSortKey mergeSortKey : sortedData) {
+      writeMergeSortKey(mergeSortKey, columnBuilders, timeColumnBuilder);
+      tsBlockBuilder.declarePosition();
+      if (tsBlockBuilder.isFull()) {
+        tsBlocks.add(tsBlockBuilder.build());
+        tsBlockBuilder.reset();
+        timeColumnBuilder = tsBlockBuilder.getTimeColumnBuilder();
+      }
+    }
+
+    if (!tsBlockBuilder.isEmpty()) {
+      tsBlocks.add(tsBlockBuilder.build());
+    }
+
+    spill(tsBlocks);
+  }
+
+  private boolean writeData(List<TsBlock> sortedData, String fileName) {
+    try {
+      Path filePath = Paths.get(fileName);
+      Files.createFile(filePath);
+
+      try (FileOutputStream fileOutputStream = new FileOutputStream(fileName)) {
+        for (TsBlock tsBlock : sortedData) {
+          ByteBuffer tsBlockBuffer = serde.serialize(tsBlock);
+          ReadWriteIOUtils.write(tsBlockBuffer, fileOutputStream);
+        }
+      }
+    } catch (Exception e) {
+      return false;
+    }
+    return true;
+  }
+
+  private void writeMergeSortKey(
+      MergeSortKey mergeSortKey, ColumnBuilder[] columnBuilders, ColumnBuilder timeColumnBuilder) {
+    timeColumnBuilder.writeLong(mergeSortKey.tsBlock.getTimeByIndex(mergeSortKey.rowIndex));
+    for (int i = 0; i < columnBuilders.length; i++) {
+      if (mergeSortKey.tsBlock.getColumn(i).isNull(mergeSortKey.rowIndex)) {
+        columnBuilders[i].appendNull();
+      } else {
+        columnBuilders[i].write(mergeSortKey.tsBlock.getColumn(i), mergeSortKey.rowIndex);
+      }
+    }
+  }
+
+  public boolean hasSpilledData() {
+    return !fileIndex.isEmpty();
+  }
+
+  public List<String> getFilePaths() {
+    List<String> filePaths = new ArrayList<>();
+    for (int index : fileIndex) {
+      filePaths.add(filePrefix + index + fileSuffix);
+    }
+    return filePaths;
+  }
+
+  public List<SortReader> getReaders(SortBufferManager sortBufferManager)
+      throws FileNotFoundException {
+    List<String> filePaths = getFilePaths();
+    List<SortReader> sortReaders = new ArrayList<>();
+    for (String filePath : filePaths) {
+      sortReaders.add(new FileSpillerReader(filePath, sortBufferManager, serde));
+    }
+    return sortReaders;
+  }
+
+  public void clear() throws IOException {
+    List<String> filePaths = getFilePaths();
+    for (String filePath : filePaths) {
+      Path newPath = Paths.get(filePath);
+      Files.deleteIfExists(newPath);
+    }
+  }
```

Comment:
No need to delete them one by one, we only need to delete one time in SortOperator(delete the corresponding folder).

Reply:
done